### PR TITLE
Spike: Use Leex as our tokenizer

### PR DESCRIPTION
**Pull request by lpil**
Very much a spike into unfamiliar territory.

https://github.com/relops/leex_yecc_example

http://cameronp.svbtle.com/how-to-use-leex-and-yecc?utm_campaign=elixir_radar_34&utm_medium=email&utm_source=RD+Station

### CODE REVIEW

File: lib/slime/lexer.ex [hunk 1]
```diff
@@ -0,0 +1,30 @@
+defmodule Slime.Lexer do
+  @moduledoc """
+  Leex based lexing of Slime documents. See src/slime_lexer.xrl for details.
+  """
+
+  def tokenize(document) when is_binary document do
```
Comment (henrik) [inline lib/slime/lexer.ex:6]: Minor: how would you feel about parentheses in `is_binary(document)` and similarly below?

Comment (henrik) [inline lib/slime/lexer.ex:6]: Heh, looking further, this is too spikey to mind these details. Sorry – carry on :)

Comment (lpil) [inline lib/slime/lexer.ex:6]: Super spikey :)

**Comment by lpil** [comment]
Closing for now.

**Pull request closed without merging**